src/frontend/staff-portal/src/app/components/ticket-info/ticket-info.component.ts

```diff
@@ -1,147 +1,575 @@
+    if (this.form.get('address').touched) return true;
+    if (this.form.get('city').touched) return true;
+    if (this.form.get('province').touched) return true;
+    if (this.form.get('postalCode').touched) return true;
+    if (this.form.get('driversLicenceNumber').touched) return true;
+    if (this.form.get('driversLicenceProvince').touched) return true;
+
+    // no contact information touched, all valid
+    return false;
   }
 
   public handleCollapse(name: string) {
-    this.collapseObj[name]= !this.collapseObj[name]
+    this.collapseObj[name] = !this.collapseObj[name]
   }
-  
-  public showManualAddress(): void {
-    this.showAddressFields = true;
+
+  // act fullsection(section)(subsection)(paragraph)
+  public getLegalParagraphing(violationTicketCount: ViolationTicketCount): string {
+    if (!violationTicketCount || !violationTicketCount.description) return "";
+    let ticketDesc = (violationTicketCount.actRegulation ? violationTicketCount.actRegulation : "") + " ";
+    if (violationTicketCount.section && violationTicketCount.section.length > 0) ticketDesc = ticketDesc + violationTicketCount.section;
+    if (violationTicketCount.subsection && violationTicketCount.subsection.length > 0) ticketDesc = ticketDesc + "(" + violationTicketCount.subsection + ")";
+    if (violationTicketCount.paragraph && violationTicketCount.paragraph.length > 0) ticketDesc = ticketDesc + "(" + violationTicketCount.paragraph + ")";
+    ticketDesc = ticketDesc + " " + violationTicketCount.description;
+    return ticketDesc;
   }
 
-  public editForm(): void {
-    this.disableForm = false;
+  // get legal paragraphing for a particular count
+  public getCountLegalParagraphing(countNumber: number, violationTicket: ViolationTicket): string {
+    if (violationTicket.violationTicketCounts.filter(x => x.count == countNumber)) return this.getLegalParagraphing(violationTicket.violationTicketCounts.filter(x => x.count == countNumber)[0]);
+    else return "";
   }
 
-  public get violationTicketNumber(): FormControl {
-    return this.form.get('violationTicketNumber') as FormControl;
+  // put dispute by id
+  putDispute(dispute: Dispute): void {
+    this.logger.log('TicketInfoComponent::putDispute', dispute);
+
+    // no need to pass back byte array with image
+    let tempDispute = dispute;
+    tempDispute.violationTicket.violationTicketImage = null;
+
+    this.busy = this.disputesService.putDispute(dispute.id, tempDispute).subscribe((response: Dispute) => {
+      this.logger.info(
+        'TicketInfoComponent::putDispute response',
+        response
+      );
+
+      // this structure contains last version of what was send to db
+      this.lastUpdatedDispute = response;
+    });
+
   }
 
-  public get violationDate(): FormControl {
-    return this.form.get('violationDate') as FormControl;
+  setViolationTicketFieldsFromJSON(dispute: Dispute): Dispute {
+    var objOcrViolationTicket = JSON.parse(dispute.ocrViolationTicket);
+
```

Comment:
please refer to the citizen portal, violation-ticket.service.ts -> function fromOCR
the json property names can be changed in backend easily, we should design which kind of naming method we should use, ticketNumber or ticket_number for all api and portal
@pbolduc @lfdgov

Reply:
i'd like to add this one as technical debt

Reply:
ok